mixbox/entities.py

```diff
@@ -466,10 +466,16 @@ def to_xml(self, include_namespaces=True, namespace_dict=None,
                 pretty_print=pretty
             )
 
-        s = six.text_type(sio.getvalue()).strip()
+        # Ensure that the StringIO buffer is unicode
+        s = six.text_type(sio.getvalue())
 
         if encoding:
-            return s.encode(encoding)
+            s = s.encode(encoding=encoding)
+            if six.PY3:
```

Comment:
If we are on Py3, shouldn't we just skip the encode/decode loop? And change the above to `if encoding and six.PY2:`? At that point, we know that `s` is a Unicode/text string (`unicode` on Py2, `str` on Py3`), so what's the reason to encode (to a byte string: `str` on Py2, `bytes` on Py3) and then decode back to text?

We also need to be clear about what we're trying to do, since this is changing the contract of `to_xml`, which up until now should have returned a byte string if you pass an encoding (or rely on the default 'utf-8' encoding), and a text string if you pass `encoding=None` (on both Python 2 and Python 3).